src/transformers/trainer_tf.py

```diff
@@ -84,10 +84,15 @@ def get_train_tfdataset(self) -> tf.data.Dataset:
         else:
             self.train_steps: int = math.ceil(self.num_train_examples / self.args.train_batch_size)
 
+        batch_size = self.args.train_batch_size
+
+        if self.args.is_tpu_available():
+            batch_size *= self.args.gradient_accumulation_steps
+
         ds = (
             self.train_dataset.cache()
             .shuffle(self.num_train_examples)
-            .batch(self.args.train_batch_size, drop_remainder=self.args.dataloader_drop_last)
+            .batch(batch_size, drop_remainder=self.args.dataloader_drop_last)
```

Comment:
Is this right ?

Batching with the whole `self.args.train_batch_size* self.args.gradient_accumulation_steps` ? I think we still want to batch on `self.args.train_batch_size`, so each training steps forward / backward only 1 batch.

Iteration on dataset is made here : https://github.com/huggingface/transformers/blob/efeb75b8054cc299698cf8bc09f395ada2660745/src/transformers/trainer_tf.py#L381 in `_accumulate_next_gradients`. 

And as we can see here : https://github.com/huggingface/transformers/blob/efeb75b8054cc299698cf8bc09f395ada2660745/src/transformers/trainer_tf.py#L354-L357 gradients are accumulated after going through `self.args.gradient_accumulation_steps` steps.

As far as I understand, we should do the batch/unbatch trick, as described in #5072 :
```python
ds = (
    self.train_dataset.cache()
    .shuffle(self.num_train_examples)
    .batch(self.args.train_batch_size * self.args.gradient_accumulation_steps, drop_remainder=self.args.dataloader_drop_last)
    .unbatch()
    .batch(self.args.train_batch_size, drop_remainder=self.args.dataloader_drop_last)
    .prefetch(tf.data.experimental.AUTOTUNE)
)
```

Reply:
I'm still not sure but ok, I trust you ^^ Will change this.

Reply:
Maybe we need the opinion of someone else :)

Reply:
I just found this https://www.kaggle.com/yihdarshieh/tpu-gradient-accumulation but the process is a bit different.